### DISCUSSION TITLE
Question about objectives text

### DISCUSSION THREAD

**Discussion by ghramsey** [MattFiler/LegendPlugin#175] [Q&A] [answered]
I have what I hope are two simple questions: are the objectives listed in the objectives file hard coded or would 
editing the text in the scipts likewise edit the dialog shown in game?  
Could one add a custom objective to say read log on terminal ?

**Comment by MattFiler** [comment]
Custom objective texts can be added yeah. I'm trying to work out where would be best to add this - perhaps you could give some thoughts. I'm thinking of either:

- In the existing localisation editor in the configuration tools (which itself I want to improve the UI of to make it easier to switch between languages)
- In the scripting tool (I'm thinking of reworking the asset suggestion dropdowns with a separate popup window which can be tailored towards each asset type to give previews, at which point, perhaps you could also add custom strings in)

Maybe in both places?

As the script editor expands it's starting to become a "perhaps a lot of these things could just be in the scripting tool" in my head, such as model/texture/material editing.

**Comment by ghramsey** [comment]
Thank you very much.  I want to make a simple objective at this point.   I added a log file from Zach Watson (the drunk guy 
who has the crowbar)  called "dead soldiers" where he's talking about getting drunk because he's bored.
I added it to a terminal, but I want something to show it's there and not TL;DR the terminal as most advanced players do. 
The best thing I can think of is an objective prompt.

I had to date never opened the localization editor. That's great!  IDK where to implement objective editing.
Maybe edit the text in the localization editor as you suggested though.

Are manditory objectives possible?  Seems the scripts would get kinda tangled and could break, but that would be wonderful 
to have it possible.

For now I'll settle for disabling unnecessary objective prompts at will such as in mission 1 when we setup Ripley to be clothed 
in her pod we still get the get dressed objective even though she is dreesed. I tried to have it not show, but I don't think I know 
what I'm doing and it didn't work.   Even current versions of KittyMod still shows "get dresed" prompt so I'm guessing it's not
 worth the time and trouble to disable.   I imagine it's not so simple as a boolean switch.   But having ability to enable/disable 
 an objective at will would be nice.

**Reply by MattFiler** [reply]
> Are manditory objectives possible? Seems the scripts would get kinda tangled and could break, but that would be wonderful
to have it possible.

Mandatory objectives are possible by hooking things like door unlocks up from the completion of an objective - that way you gate players in a space until they've completed the objective! Use `SetPrimaryObjective` for this - you can set sub objectives too which are the tasks that get ticked off in the tab menu.

**Comment by ghramsey** [comment]
OH.  I forgot to ask about word wrapping. I changed the tape in the bar in M2 with Julia Jones' alternate dialog. 
This one [AI_UI_JUL_9463_ALT].  The word wrapping is different than normal and scrolling oddly. 
Any reason for this?  Is there a width limitation where the game doesn't automatically word wrap the text?

**Reply by MattFiler** [reply, marked as answer]
I don't think the text is auto word-wrapped - I think you have to define the line lengths in the string? Not 100% on that one though.
